### PR TITLE
feat: FeeRebate: finalising FeeRebateHook per design

### PR DESCRIPTION
**Pull request by davidprepo**
FeeRebateHook was missing some implementation and was still inheriting access control.

See docs here: https://www.notion.so/prepo/Fee-Rebate-SC-Spec-5723bb91a17c42e2a409ffd9af591085#fa49c2a60fcf4af5968c5a7f6ae02380

Following the existing style in this file (ie internal _setter with external setter).

### CODE REVIEW

File: apps/smart-contracts/core/contracts/FeeRebateHook.sol [hunk 1]
```diff
@@ -22,9 +21,16 @@ contract FeeRebateHook is IFeeRebateHook, SafeAccessControlEnumerable {
...
-    override
-  {}
+  function setTokenSender(ITokenSender tokenSender) external virtual override {
+    _setTokenSender(tokenSender);
+  }
+
+  function _setTokenSender(ITokenSender tokenSender) internal {
+    _feeRebate = tokenSender;
+    emit TokenSenderChange(address(tokenSender));
+  }
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/contracts/FeeRebateHook.sol:31]: Can remove this and just make `setTokenSender`, `public`.

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides pending name change? If we haven't decided on one by the time you finish making the edits, we can rename it later

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

File: apps/smart-contracts/core/contracts/FeeRebateHook.sol [hunk 2]
```diff
@@ -3,17 +3,12 @@ pragma solidity =0.8.7;
 
 import "./interfaces/IFeeRebateHook.sol";
 import "./interfaces/ITokenSender.sol";
-import "prepo-shared-contracts/contracts/SafeAccessControlEnumerable.sol";
 
-contract FeeRebateHook is IFeeRebateHook, SafeAccessControlEnumerable {
+contract FeeRebateHook is IFeeRebateHook {
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/contracts/FeeRebateHook.sol:7]:
```suggestion
contract TokenSenderCaller is ITokenSenderCaller {
```
Rename as per discord discussion, feel free to apply any renames within this PR.

Comment (davidprepo) [reply apps/smart-contracts/core/contracts/FeeRebateHook.sol:7]: Will do in separate PR after train is merged.

**Pull request merged**